torchvision/datasets/utils.py

```diff
@@ -83,7 +83,7 @@ def download_url(url, root, filename=None, md5=None):
                 url, fpath,
                 reporthook=gen_bar_updater()
             )
-        except urllib.error.URLError as e:
+        except (urllib.error.URLError, IOError) as e:
```

Comment:
Can you explain why this change was needed?

From https://stackoverflow.com/questions/29347790/difference-between-ioerror-and-oserror, `IOError` is (in Python3) an alias for `OSError`, and [`URLError`](https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError) is a subclass of `IOError` in Py2 and of `OSError` in Python3.

Reply:
On Python 2.7 I observed that a certificate validation failure results in an IOError, not a URLError:

```

    def do_handshake(self, block=False):
        """Perform a TLS/SSL handshake."""
        self._check_connected()
        timeout = self.gettimeout()
        try:
            if timeout == 0.0 and block:
                self.settimeout(None)
>           self._sslobj.do_handshake()
E           IOError: [Errno socket error] [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
```

Reply:
But from https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError, `URLError` is a subclass of `IOError` in Python 2 . Was it failing?

Reply:
Oh, I see, forget what I said.